### PR TITLE
VPN-3511 Remove: GetTransportType Code

### PR DESCRIPTION
**Pull request by Gela**
## Description

Remove `GetTransportTypeCode` and replace with `getReachability()` instead. GetTransportTypeCode became obsolete with Qt 6.3.

## Reference

    https://mozilla-hub.atlassian.net/browse/VPN-3511

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: src/platforms/android/androidnetworkwatcher.cpp [hunk 1]
```diff
@@ -16,8 +17,8 @@
 namespace {
 Logger logger("AndroidNetworkWatcher");
 
-constexpr auto VPNNetworkWatcher_CLASS =
-    "org/mozilla/firefox/vpn/qt/VPNNetworkWatcher";
+// constexpr auto VPNNetworkWatcher_CLASS =
+//     "org/mozilla/firefox/vpn/qt/VPNNetworkWatcher";
```
Comment (brizental) [inline src/platforms/android/androidnetworkwatcher.cpp:21]: Why not just remove this instead?

File: src/telemetry.cpp [hunk 2]
```diff
@@ -120,17 +120,17 @@ void Telemetry::initialize() {
...
+  connect(
+      controller, &Controller::handshakeFailed, this,
+      [](const QString& publicKey) {
+        logger.info() << "Send a handshake failure event";
+        mozilla::glean::sample::connectivity_handshake_timeout.record(
+            mozilla::glean::sample::ConnectivityHandshakeTimeoutExtra{
+                ._server = publicKey,
+                ._transport = QVariant::fromValue(MozillaVPN::instance()
+                                                      ->networkWatcher()
+                                                      ->getReachability())});
```
Comment (brizental) [inline src/telemetry.cpp:132]: Please update the documentation for this metric. 

Note that usually this would not be an ok change -- changing values in telemetry, but since we are not looking at these metrics in any dashboard that I know of we can just go forward.

File: src/telemetry.cpp [hunk 3]
```diff
@@ -258,7 +258,8 @@ void Telemetry::connectionStabilityEvent() {
           ._loss = QString::number(vpn->connectionHealth()->loss()),
           ._server = vpn->controller()->currentServer().exitServerPublicKey(),
           ._stddev = QString::number(vpn->connectionHealth()->stddev()),
-          ._transport = vpn->networkWatcher()->getCurrentTransport()});
+          ._transport =
+              QVariant::fromValue(vpn->networkWatcher()->getReachability())});
```
Comment (brizental) [inline src/telemetry.cpp:262]: Same comment as the other telemetry change.

**Review by brizental — APPROVED** [review]

File: src/telemetry/metrics.yaml [hunk 4]
```diff
@@ -1111,8 +1111,8 @@ sample:
           loss, stddev.
         type: string
       transport:
-        description: The Host network type (i.e wifi or 4g)
-        type: string
+        description: The Host network type (i.e wifi or 4g). Possible values are Unknown, Disconnected, Local, Site, Online
+        type: QNetworkInformation::Reachability
```
Comment (brizental) [inline src/telemetry/metrics.yaml:1115]:
```suggestion
        type: string
```

File: src/telemetry/metrics.yaml [hunk 5]
```diff
@@ -1148,8 +1148,8 @@ sample:
         description: Standard deviation of the ping responses
         type: string
       transport:
-        description: The Host network type (i.e wifi or 4g)
-        type: string
+        description: The Host network type (i.e wifi or 4g). Possible values are Unknown, Disconnected, Local, Site, Online
+        type: QNetworkInformation::Reachability
```
Comment (brizental) [inline src/telemetry/metrics.yaml:1152]: string
```suggestion
        type: string
```

File: src/telemetry/metrics.yaml [hunk 6]
```diff
@@ -1148,8 +1148,8 @@ sample:
         description: Standard deviation of the ping responses
         type: string
       transport:
-        description: The Host network type (i.e wifi or 4g)
-        type: string
+        description: The Host network type (i.e wifi or 4g). Possible values are Unknown, Disconnected, Local, Site, Online
```
Comment (brizental) [inline src/telemetry/metrics.yaml:1151]:
```suggestion
        description: |
             The Host network type (i.e wifi or 4g). 
             Possible values are Unknown, Disconnected, Local, Site, Online
```
